src/server/game/BattlePets/BattlePetMgr.cpp

```diff
@@ -677,6 +678,72 @@ void BattlePetMgr::ChangeBattlePetQuality(ObjectGuid guid, BattlePetBreedQuality
     // _owner->GetPlayer()->SetCurrentBattlePetBreedQuality(qualityValue);
 }
 
+void BattlePetMgr::GrantBattlePetExperience(ObjectGuid guid, uint16 xp)
```

Comment:
This should have another argument `xpSource` that will let you determine if aura bonus should be applied or if CriteriaType::ActivelyEarnPetLevel should be updated (level boosts shouldnt)

Reply:
Added the new argument (enum class BattlePetXpSource)